jenkins/clean-jenkins-bots.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash -ex
+
+# Do some simple validation
+if [[ "$BUILD_REVISION" != "jenkins" ]] ; then
+	echo "This script should only be run on Jenkins bots."
+	exit 1
+fi
+
+# Print disk status before cleaning
+df -h
+
+# We don't care about errors in this section, we just want to clean as much as possible
+set +e
+
+# Delete all the simulator devices. These can take up a lot of space over time (I've seen 100+GB on the bots)
+/Applications/Xcode.app/Contents/Developer/usr/bin/simctl delete all
+
+# Delete old Xcodes.
+ls -lad /Applications/Xcode*.app
+sudo rm -Rf /Applications/Xcode44.app
```

Comment:
Is there a reason for each of these to be their own `sudo rm` instead of
```
oldXcodes=(
   "/Applications/Xcode5.app"
   "/Applications/Xcode502.app"
   # etc
}

for oldXcode in ${oldXcodes[@]}; do
    sudo rm -Rf $oldXcode
done
```

Reply:
I went that path in the yaml version: https://github.com/xamarin/xamarin-macios/blob/main/tools/devops/device-tests/scripts/System.psm1#L188

Reply:
> Is there a reason for each of these to be their own `sudo rm` instead of

Not really, so I changed it.